src/test/java/htsjdk/variant/variantcontext/writer/VariantContextWriterBuilderUnitTest.java

```diff
@@ -406,4 +481,50 @@ public void testStdOut() {
         Assert.assertNotNull(((VCFWriter) writer).getOutputStream());
         Assert.assertNotEquals(((VCFWriter) writer).getStreamName(), IndexingVariantContextWriter.DEFAULT_READER_NAME);
     }
+
+  /**
+   * Create a named pipe, call "codeThatWritesToAFilename" with the name as argument,
+   * consumes the output as it's being run and returns the number of bytes that were output.
+   *
+   * This only works on Unix. Use this decoration for the tests that use it:
+   * @Test(singleThreaded = true, groups = "unix")
+   */
+    private long testWriteToPipe(final Consumer<String> codeThatWritesToAFilename) throws IOException, InterruptedException, ExecutionException {
+      // make the fifo
+      final File fifo = File.createTempFile("fifo.for.testWriteToPipe", "");
+      Assert.assertTrue(fifo.delete());
+      fifo.deleteOnExit();
+      final Process exec = new ProcessBuilder("mkfifo", fifo.getAbsolutePath()).start();
+      exec.waitFor(1, TimeUnit.MINUTES);
+      Assert.assertEquals(exec.exitValue(), 0, "mkfifo failed with exit code " + 0);
+
+      // run the code in a separate thread
+      ExecutorService executor = null;
+      try {
+        executor = Executors.newSingleThreadExecutor();
```

Comment:
move this out of the try and then you can set the executor to final and remove the if statement in 526